### PR TITLE
Add classification task skeleton

### PR DESCRIPTION
**Pull request by lewfish**
This PR adds a skeleton for the classification MLTask. It also changes some terminology to be more consistent with other projects (annotation -> label and label->class).

### CODE REVIEW

File: src/rv2/core/labels.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+
+class Labels(ABC):
+    '''A set of spatially referenced labels.
+
+    A set of labels associated with their location predicted by a model or
+    provided by human labelers for the sake of training. An example of an label
+    is a bounding box in the case of object detection or a raster in case of
+    segmentation.
+    '''
```
Comment (lossyrob) [inline src/rv2/core/labels.py]: Just to be clear: for classification, the classes are still spatially referenced, by tying the chip extent to the classification(s)?

File: src/rv2/ml_backends/keras_classification.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from rv2.core.ml_backend import MLBackend
+
+
+class KerasClassification(MLBackend):
```
Comment (lossyrob) [inline src/rv2/ml_backends/keras_classification.py]: Are Backends many to one to Tasks? i.e. there wouldn't be a generic "Keras" backend that could do multiple tasks, that a Backend only can perform one task?

File: src/rv2/core/labels.py (hunk 1 above)
Comment (lewfish) [reply src/rv2/core/labels.py]: Yes. I'll clarify this in the comment.

File: src/rv2/ml_backends/keras_classification.py (hunk 2 above)
Comment (lewfish) [reply src/rv2/ml_backends/keras_classification.py]: Yes.

Comment (lewfish) [reply src/rv2/ml_backends/keras_classification.py]: Updating comments.

**Comment by lossyrob** [comment]
👍 lgtm - one comment is that separating out the renames with the skeleton would have been useful, as to have a reference PR for adding the necessary pieces for adding a backend and task. Might be more effort than worth to do now, but something to think of in a future PR/set of PRs.

**Comment by lewfish** [comment]
> one comment is that separating out the renames with the skeleton would have been useful, as to have a reference PR for adding the necessary pieces for adding a backend and task. Might be more effort than worth to do now, but something to think of in a future PR/set of PRs.

True, but it's in its own commit which should still be helpful.

**Pull request merged**